update install guide to inject binaries for argcomplete, fixes #184

marcolussetti opened the pull request:
By default, pipx does not include binaries for injected dependencies in the path.

So, the command suggested a few moments after, `activate-global-python-argcomplete` will not work as that binary is not in the path.

By telling the pipx inject command to `--include-binaries`, it will be added to the path.

marcolussetti committed:
update install guide to inject binaries for argcomplete

By default, pipx does not include binaries for injected dependencies in the path.

So, the command suggested a few moments after, `activate-global-python-argcomplete` will not work as that binary is not in the path.

By telling the pipx inject command to `--include-binaries`, it will be added to the path.

marcolussetti commented:
This fixes issue #184 (which I have just filed).

gotmax23 commented on docs/docsite/rst/installation_guide/intro_installation.rst:
```diff
@@ -99,7 +99,7 @@ To install additional python dependencies that may be needed, with the example o
 .. code-block:: console
 
-    $ pipx inject ansible argcomplete
+    $ pipx inject --include-binaries ansible argcomplete
```
`--include-binaries` doesn't exist. I think you want `--include-apps`.

marcolussetti replied:
Oh I see. Thank you for catching that. Looks like this is dependent on the version of `pipx` unfortunately, and I was working off Ubuntu 20.04 which ships an older version of pipx. `--include-binaries` was replaced in 0.14.0.0 by `--include-apps`.

I will update.

gotmax23 replied:
You switched the wrong line. `pipx install --include-deps ansible` was correct. `pipx inject --include-binaries ansible argcomplete` was not.

marcolussetti replied:
Oh dear, that is quite embarrassing!

 I'm sorry, I think I need to get some more tea this morning.

marcolussetti committed:
fix intro guide pipx inject based on pr feedback

0.14.0 of pipx switched to --include-apps instead of --include-binaries

marcolussetti committed:
undo incorrect commit, expand explanation

pipx after version 0.14.0 renamed --include-binaries to --include-apps

Additionally, it seems more appropriate to have that call to
--include-apps in the argcomplete-specific section rather than in the
section for how to inject dependencies in general.

marcolussetti commented:
@gotmax23 : I realized that probably the right place to place the `--include-apps` parameter is in the section specific to `argcomplete`. However, given the example is also used in the "Installing Extra Python Dependencies", some explanation should probably be provider there.

I have attempted to do so, not too sure how clear my explanation is, admittedly.

samccann commented:
Please don't merge until we have approval from core folks. I've added @sivel in the review requests...

marcolussetti commented:
(I think the conversation was resolved, but it was still sitting in requested changes)

oraNod commented on docs/docsite/rst/installation_guide/intro_installation.rst:
```diff
@@ -101,6 +101,12 @@ To install additional python dependencies that may be needed, with the example o
 
     $ pipx inject ansible argcomplete
 
+If the additional python dependency needed includes commands that need to be executed on their own from the shell, `--include-apps` should be included to make those commands accessible.
```
```suggestion
Include the ``--include-apps`` option to make apps in the additional python dependency available on your PATH. This allows you to execute commands for those apps from the shell.
```

felixfontein replied:
```suggestion
If the additional python dependency needed includes commands that need to be executed on their own from the shell, ``--include-apps`` should be included to make those commands accessible.
```

samccann committed:
Update docs/docsite/rst/installation_guide/intro_installation.rst

Co-authored-by: Don Naro <dnaro@redhat.com>
